include/fplus/string_tools.hpp

```diff
@@ -125,6 +125,28 @@ String trim_whitespace(const String& str)
     return trim_by(is_whitespace<String>, str);
 }
 
+// API search type: to_lower_case : String -> String
+// to_lower_case("ChaRacTer&WorDs23") == "character&words23"
+template <typename String>
+std::string to_lower_case( const std::string& str ) {
+    return fplus::transform( [](char c) -> char{
```

Comment:
Was there no way to use `std::tolower`? http://en.cppreference.com/w/cpp/string/byte/toupper